File: problems/数学与位运算/191.位1的个数.cpp
#include <iostream>
#include <cstdint>
using namespace std;

int hammingWeight(uint32_t n) {
    int count = 0;
    while (n) { n &= n - 1; count++; }
    return count;
}

int main() {
    // 测试用例
    cout << hammingWeight(11) << endl;          // 预期: 3
    cout << hammingWeight(128) << endl;         // 预期: 1
    cout << hammingWeight(2147483647u) << endl; // 预期: 31
    cout << hammingWeight(0) << endl;            // 预期: 0
    return 0;
}

File: problems/数学与位运算/191.位1的个数.py
#!/usr/bin/env python3
"""
191. 位1的个数
"""
def hamming_weight(n: int) -> int:
    count = 0
    while n:
        n &= n - 1
        count += 1
    return count

if __name__ == "__main__":
    # 测试用例
    print(hamming_weight(11))  # 预期: 3  (二进制: 1011)
    print(hamming_weight(128))  # 预期: 1  (二进制: 10000000)
    print(hamming_weight(2147483647))  # 预期: 31 (二进制: 31个1)
    print(hamming_weight(0))  # 预期: 0
